### DISCUSSION TITLE
Streamlink CLI won't allow recording to path with a period in the folder name

### DISCUSSION THREAD

**Discussion by SeraphSephiroth** [streamlink/streamlink#3664] [Q&A]
So I've been writing a script to record and sort things, and so far things are going okay, but I've run into a problem.

The code I use creates a subdirectory in the folder with the channel name (generated by ytdl), and is meant to record the stream into that folder.

However, if the channel name has a . (period) in it (Such as hololive channels etc), and the folder is generated with that name, streamlink will refuse to record into that folder, even if the path is in quotation marks (to denote a path with spaces).

For instance:

Channel Name: Watson Amelia Ch. hololive-EN

Script creates folder \Recordings\Watson Amelia Ch. hololive-EN

Script runs the recording command: 
`--retry-streams 10 -o "$scriptdir\Recordings\$channel\stream.mp4" $streamurl best`

However, when trying to record, streamlink exits with the following error:
`streamlink` args:  --retry-streams 10 -o W:\recordings\Recordings\Watson Amelia Ch. hololive-EN\stream.mp4 https://www.youtube.com/watch?v=EeC6OHBRFTc best
usage: streamlink [OPTIONS] <URL> [STREAM]
streamlink: error: unrecognized arguments: hololive-EN\stream.mp4 https://www.youtube.com/watch?v=EeC6OHBRFTc best`

Is there something I'm doing wrong here? Or is it the . that's causing the issue?

(Edit: You can workaround by moving into that channel folder and running:
`--retry-streams 10 -o stream.mp4 $streamurl best`
But that's not ideal really)

Any help would be appreciated.

**Comment by gravyboat** [comment]
Your Streamlink command:
```
streamlink --retry-streams 10 -o W:\recordings\Recordings\Watson Amelia Ch. hololive-EN\stream.mp4 https://www.youtube.com/watch?v=EeC6OHBRFTc best
```
 isn't wrapping the output path in quotes correctly. Streamlink is working perfectly fine:
```
streamlink --retry-streams 10 -o "C:\test\path\name. test\stream.mp4" https://www.youtube.com/watch?v=EeC6OHBRFTc best
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=EeC6OHBRFTc
[cli][info] Available streams: audio_mp4a, audio_opus, 144p (worst), 240p, 360p, 480p, 720p, 720p60, 1080p60 (best)
[cli][info] Opening stream: 1080p60 (muxed-stream)
[download][stream.mp4] Written 7.8 MB (1s @ 4.8 MB/s)                                                                  [cli][info] Stream ended
Interrupted! Exiting...
[cli][info] Closing currently open stream...
```

Your command needs to look like this when it is executed:
```
streamlink --retry-streams 10 -o "W:\recordings\Recordings\Watson Amelia Ch. hololive-EN\stream.mp4" https://www.youtube.com/watch?v=EeC6OHBRFTc best
```

Without wrapping the whole path properly (looks like your script is using the quotes as variable substitution) Streamlink will interpret the space as another option.

**Reply by SeraphSephiroth** [reply]
The thing is I use the variables to denote paths.
$scriptdir = Script location
$channel = Youtube channel name

The script won't work with a full path unless I want to type it out every time.

I've just tried again, wrapping the path properly, and something else seems to be wrong.

`streamlink args:  --retry-streams 10 -o W:\recordings\Recordings\Lamy Ch. \stream.mp4 https://www.youtube.com/watch?v=9kxjbaQOFo0 best
usage: streamlink [OPTIONS] <URL> [STREAM]
streamlink: error: unrecognized arguments: https://www.youtube.com/watch?v=9kxjbaQOFo0 best`

Command: 
`streamlink --retry-streams 10 -o "$scriptdir\Recordings\$channel\stream.mp4" $streamurl best`

Running this:
`cd $scriptdir\Recordings\$channel\`
`streamlink --retry-streams 10 -o stream.mp4 $streamurl best`

works fine.

**Reply by gravyboat** [reply]
You're not wrapping the path properly based on your command. I see no quotes around the actual path when the command is run.

**Reply by SeraphSephiroth** [reply]
I don't know why. The script itself has the quotes around the output path, so in theory it should place them when the script is being run. Is that a powershell thing? Something isn't being passed right...

`streamlink --retry-streams 10 -o "$scriptdir\Recordings\$channel\stream.mp4" $streamurl best`

This is the command, and it looks okay to me. Is there a different way to wrap this path that I don't know? The variables need to be used, so I can't change those.

**Reply by gravyboat** [reply]
You'll have to Google that, it's outside the level of support we can provide. Streamlink is working fine, you need to make sure everything is being passed through properly.

**Reply by SeraphSephiroth** [reply]
Okie dokie. Thanks for the help.
